ledger-core-algorand/src/algorand/AlgorandAccount.cpp

```
@@ -0,0 +1,218 @@
+    }
+
+    namespace {
+
+        void setAmountAndRecipients(model::Operation& op,
+                                    const model::Transaction& tx)
+        {
+            if (tx.header.type == model::constants::pay) {
+                const auto& details = boost::get<model::PaymentTxnFields>(tx.details);
+                op.amount = BigInt(static_cast<unsigned long long>(details.amount));
+                op.recipients = {};
+                op.recipients.push_back(details.receiverAddr.getAddress());
+                if (details.closeAddr) {
+                    op.recipients.push_back(details.closeAddr->getAddress());
+                    // TODO : adjust amount in this case?
+                }
+            } else if (tx.header.type == model::constants::axfer) {
+                const auto& details = boost::get<model::AssetTransferTxnFields>(tx.details);
+                op.amount = BigInt(static_cast<unsigned long long>(details.assetAmount));
+                op.recipients = {};
+                op.recipients.push_back(details.assetReceiver.getAddress());
+                if (details.assetCloseTo) {
+                    op.recipients.push_back(details.assetCloseTo->getAddress());
+                    // TODO : adjust amount in this case?
+                }
+            }
+        }
+
+    } // namespace
+
+    void Account::inflateOperation(model::Operation& op,
+                                   const std::shared_ptr<const AbstractWallet>& wallet,
+                                   const model::Transaction& tx)
+    {
+        op.setTransaction(tx);
+        op.accountUid = getAccountUid();
+        op.walletUid = wallet->getWalletUid();
+        op.date = {}; // TODO (when date is available in the tx)
+        op.senders = { tx.header.sender.getAddress() };
+        setAmountAndRecipients(op, tx);
+        op.fees = BigInt(static_cast<unsigned long long>(tx.header.fee));
+        op.block; // TODO: set or ignore?
```

Good question... does the Live need block information as part of the Operations?... Probably not...
However you can at least set these 3 fields since we have them:

- `api::Block::time` (when available with the transaction)
- `api::Block::currencyName`
- `api::Block::height`

---

I think api::Block::height is an int64_t, should we raise an exception if we overflow, as we are using uint64_t?

---

Ach! I agree, that's an issue I also encountered in `ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp` lines 150-154 here : https://github.com/Ledger-Coin-Integration-team/lib-ledger-core/pull/14/files
Not sure it's the best one but that's how I got around it...